headers/Scene.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+class Application;
+
+class Scene {
+	protected:
+		Application* app;
+	public:
+		Scene(Application* app);
+		virtual void OnCreate() = 0;
+		virtual void OnUpdate() = 0;
```

Comment:
We'll want to pass `elapsedTime` to the update method so updates are based on time and not frames.
```suggestion
		virtual void OnUpdate(Uint32 elapsedTime) = 0;
```